phpBB/includes/functions_admin.php

```diff
@@ -2573,6 +2573,11 @@ function prune($forum_id, $prune_mode, $prune_date, $prune_flags = 0, $auto_sync
 		$sql_and .= ' AND topic_status = ' . ITEM_MOVED . " AND topic_last_post_time < $prune_date";
 	}
 
+	if ($prune_limit > 0)
+	{
+		$sql_and .= " LIMIT $prune_limit";
```

Comment:
I don't think we'll approach a prune limit this high so I do still think that using sql_query_limit() should be ok here and also be the better way. (better as in: the DBAL should handle the limit. some DBALs, maybe even future supported types, might handle the limit differently).
I'll try to write something about the events when I'm back home. ;)